src/extension/features/general/colour-blind-mode/index.js

```diff
@@ -1,7 +1,356 @@
+      document.body.removeAttribute(`tk-colour-blind-${name}-square`);
+    }
+  }
+
+  getSquare(name) {
+    return !!document.body.getAttribute(`tk-colour-blind-${name}-square`);
+  }
+
+  saveSquare(name) {
+    setToolkitStorageKey(`colour-blind-${name}-square`, this.getSquare(name));
+  }
+
+  loadSquare(name, def) {
+    return getToolkitStorageKey(`colour-blind-${name}-square`, def);
+  }
+
+  loadSettings() {
+    // Load colours from storage - otherwise use values from style sheet
+    var positive = this.loadColour('positive', this.getColour('positive'));
+    var warning = this.loadColour('warning', this.getColour('warning'));
+    var negative = this.loadColour('negative', this.getColour('negative'));
+
+    this.setColour('positive', positive);
+    this.setColour('warning', warning);
+    this.setColour('negative', negative);
+
+    this.setSquare('positive', this.loadSquare('positive', false));
+    this.setSquare('warning', this.loadSquare('warning', false));
+    this.setSquare('negative', this.loadSquare('negative', true)); // On by default
+  }
+
+  saveSettings() {
+    this.saveColour('positive');
+    this.saveColour('warning');
+    this.saveColour('negative');
+
+    this.saveSquare('positive');
+    this.saveSquare('warning');
+    this.saveSquare('negative');
+  }
+
+  cancelChanges() {
+    // Reset the live colour preview - this is necessary because if the user has
+    // no saved settings the default colour otherwise can't be read
+    this.resetColour('positive');
+    this.resetColour('warning');
+    this.resetColour('negative');
+
+    this.loadSettings();
+  }
+
+  invoke() {
+    this.loadSettings();
+  }
+
+  createColourOption(name, label) {
+    var value = this.getColour(name);
+
+    var button = $(
+      `<button>
+        <div class="tk-colour-blind-${name}">
+          <input type="color" value="${value}" class="tk-colour-blind-picker" style="background-color: ${value};"></input>
+        </div>
+        <div class="ynab-new-theme-switcher-label">${label}</div>
+      </button>`
+    );
+
+    // Live update colours as they are selected
+    $('input', button).on('input', (e) => {
+      this.setColour(name, e.target.value);
+    });
+
+    // If the user clicks on the button, redirect it to the input
+    button.on('click', (e) => {
+      let input = $('input', button);
+      if (e.target !== input.get(0)) {
+        input.trigger('click');
+      }
+    });
+
+    return button;
+  }
+
+  createSquareOption(name) {
+    var square = this.getSquare(name);
+
+    var option = $(
+      `<div>
+        <input type="checkbox" id="tk-colour-blind-${name}-square" ${square ? 'checked' : ''}>
+        <label for="tk-colour-blind-${name}-square">Square Corners</label>
+      </div>`
+    );
+    $('input', option).on('click', (e) => {
+      this.setSquare(name, e.target.checked);
+    });
+    return option;
+  }
+
+  buildOptionsMenu() {
+    let themeSwitcher = $(`.${YNAB_THEME_SWITCHER}`);
```

Comment:
can we also add a button to "reset to default" that way users have a way back when they inevitably pick a color they regret :D

Reply:
Absolutely, I've been meaning to get to it - if there's space a dropdown with a couple of different defaults might even be worth doing. I'll sort out a reset button for the time being though until I get a chance to figure that out.